pay/contracts/subscription/HectorSubscription.sol

```diff
@@ -0,0 +1,498 @@
+        }
+    }
+
+    function refund(
+        address[] memory _tos,
+        address[] memory _tokens,
+        uint256[] memory _amounts
+    ) external onlyMod {
+        uint256 length = _tos.length;
+        if (length != _tokens.length) revert INVALID_PARAM();
+        if (length != _amounts.length) revert INVALID_PARAM();
+
+        for (uint256 i = 0; i < length; i++) {
+            address to = _tos[i];
+
+            /// only active subscription
+            if (subscriptions[to].planId == 0) {
+                continue;
+            }
+
+            address token = _tokens[i];
+            uint256 amount = _amounts[i];
+
+            if (token == address(0)) revert INVALID_ADDRESS();
+            if (amount == 0) revert INVALID_AMOUNT();
+
+            IERC20(token).safeTransfer(to, amount);
+
+            emit Refunded(to, token, amount);
+        }
+    }
+
+    /* ======== VIEW FUNCTIONS ======== */
+
+    function allPlans() external view returns (Plan[] memory) {
+        return plans;
+    }
+
+    function getSubscription(address from)
+        external
+        view
+        returns (
+            uint256 planId,
+            uint48 expiredAt,
+            bool isCancelled,
+            bool isActiveForNow,
+            uint256 chargeAmount
+        )
+    {
+        Subscription memory subscription = subscriptions[from];
+        planId = subscription.planId;
+        expiredAt = subscription.expiredAt;
+
+        // cancelled subscription
+        if (planId == 0) {
+            return (0, expiredAt, true, block.timestamp < expiredAt, 0);
+        }
+
+        // before expiration
+        if (block.timestamp < expiredAt) {
+            isActiveForNow = true;
+        } else {
+            // after expiration
+            Plan memory plan = plans[planId];
+            uint256 count = (block.timestamp - expiredAt + plan.period) /
+                plan.period;
+            uint256 amount = plan.amount * count;
+            uint256 balance = balanceOf[from][plan.token];
+
+            if (balance >= amount) {
+                isActiveForNow = true;
+                chargeAmount = 0;
+            } else {
+                isActiveForNow = false;
+                chargeAmount = amount - balance;
+            }
+        }
+    }
+
+    /* ======== USER FUNCTIONS ======== */
+
+    function deposit(address _token, uint256 _amount) public nonReentrant {
+        balanceOf[msg.sender][_token] += _amount;
+
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
+
+        emit PayerDeposit(msg.sender, _token, _amount);
+    }
+
+    function createSubscription(uint256 _planId) public onlyValidPlan(_planId) {
+        if (subscriptions[msg.sender].planId > 0) revert ACTIVE_SUBSCRIPTION();
+
+        Plan memory plan = plans[_planId];
+
+        // Pay first plan
+        if (balanceOf[msg.sender][plan.token] < plan.amount)
+            revert PAYER_IN_DEBT();
```

Comment:
@bejaxer maybe the error  "INSUFFICIENT_FUND" would be more accurate than PAYER_IN_DEBT